src/main/java/dog/pawbook/logic/commands/AddDogCommand.java

```diff
@@ -52,6 +58,28 @@ protected String getDuplicateMessage() {
         return MESSAGE_DUPLICATE_DOG;
     }
 
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (model.hasEntity(toAdd)) {
+            throw new CommandException(getDuplicateMessage());
+        }
+
+        Owner o = (Owner) model.getFilteredEntityList().stream()
+            .filter(p -> p.getKey() == toAdd.getOwnerId())
+            .findFirst().orElseThrow()
+            .getValue();
```

Comment:
is it better to use the `hasEntity` method to check if the ID exists and throw an exception if it doesn't. then we can assume that `getFilteredEntityList` will return something that definitely contains the owner, then we retrieve it for real this time and check if it is indeed an owner object instead of any other entities. It will be better to use the `setEntity` method to change the target owner by creating a new copy with the up to date details.

Reply:
Another thing of note is that we have to ensure that the check that the entity we have is an instance of `Owner` first, or else we might be downcasting illegally, it would be a good opportunity to throw an exception along the lines of id doesn't refer to owner.